doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,146 @@
+.. _demeshening:
+
+How Particles are Indexed
+=========================
+
+With yt-4.0, the method by which particles are indexed changed considerably.
+Whereas in previous versions, particles were indexed based on their position in
+an octree (the structure of which was determined by particle number density),
+in yt-4.0 this system was overhauled to utilize a `bitmap
+index<https://en.wikipedia.org/wiki/Bitmap_index>`_ based on a space-filling
+curve, using a `enhanced word-aligned
+hybrid<https://github.com/lemire/ewahboolarray>` boolean array as their
+backend.
+
+.. note::
+
+   You may see scattered references to "the demeshening" (including in the
+   filename for this document!)  This was a humorous name used in the yt
+   development process to refer to removing a global (octree) mesh for
+   particle codes.
```

Comment:
Punctuation :)
Also, really cool to have this note, I was going to comment that maybe the word "demeshening" was probably note really to pass the test of time very well, seeing that it's mostly historic, but it makes sense to have it documented as such, at least for a few years !
```suggestion
   You may see scattered references to "the demeshening" (including in the
   filename for this document!). This was a humorous name used in the yt
   development process to refer to removing a global (octree) mesh for
   particle codes.
```